src/Compilers/Test/Core/Assert/AssertEx.cs

```diff
@@ -160,6 +161,12 @@ public static void AreEqual<T>(T expected, T actual, string message = null, IEqu
             }
         }
 
+        public static void Equal<T>(ReadOnlySpan<T> expected, T[] actual) =>
+            Equal<T>(expected.ToArray(), actual);
```

Comment:
Nit: since we know allocations are a problem in tests, perhaps instead call `Equal<T>(expected, actual.AsSpan())`?